### DISCUSSION TITLE
4 sensors (like a box) following the raycaster position to detect collisions with walls etc.

### DISCUSSION THREAD

**Discussion by rossmarinaro** [enable3d/enable3d#262] [Help / Questions] [answered]
I am wondering how I can acheive collisions within a model, like a level. So far I've detected the collisions with intersectObject, but what I can't figure out is how to get the sensors (just physics boxes) to follow me around and always have the proper rotation. I can get them to move with the camera, just not properly stay in place, encasing the first person camera so to speak.

here is one sensor, there are three others all forming a box around the player

        this.sensorUp = this.third.physics.add.box({name: 'sensorUp', x: 5, y: 5, z: 20, width: 15, height: 15, depth: 2, lambert: {color: 
       0xff0000}, collisionFlags: 2});

     this.sensorUp.position.set(player.position.x, player.position.y, player.position.z - 20) 
      this.sensorUp.rotation.copy(this.third.camera.rotation);
      this.sensorUp.body.needUpdate = true;

**Comment by yandeu** [comment, marked as answer]
I'm not 100% sure what you mean.

But I use a sensor in [this example](https://enable3d.io/examples/first-phaser-game-3d-version.html) to detect if the player is on a platform. Maybe it helps?

**Reply by rossmarinaro** [reply]
yeah that example is great, I was following this, however I'm looking to use sensors with the first person camera, so that if the front is colliding, it cancels the movement. If the left sensor is colliding, left movement is canceled etc. My problem is I'm not sure how to get the sensors to stay in position, and follow the camera (player view) around.

**Reply by yandeu** [reply]
I guess you have to sync it manually. You can use a ghost and kinematic body if you set the collisionFlags to 6 (2 + 4).

**Reply by rossmarinaro** [reply]
Actually, sync 2d to 3d seems to be the answer in this situation, so that the sensor follows along a proper axis of the screen.
